src/ucp/wireup/wireup.c

```diff
@@ -642,10 +642,16 @@ static ucs_status_t ucp_wireup_msg_handler(void *arg, void *data,
     ucp_worker_h worker   = arg;
     ucp_wireup_msg_t *msg = data;
     ucp_unpacked_address_t remote_address;
+    ucp_ep_h ep UCS_V_UNUSED;
     ucs_status_t status;
 
     UCS_ASYNC_BLOCK(&worker->async);
 
+    if (msg->dest_ep_ptr != 0) {
+        ep = ucp_worker_get_ep_by_ptr(worker, msg->dest_ep_ptr);
+        ucs_assert(ucp_ep_get_cm_lane(ep) == UCP_NULL_LANE);
+    }
```

Comment:
why needed? we're planning to expand wireup for CM case as well to send extra transport info

Reply:
but today this flow is impossible, I see 2 reasons for adding:
1) to catch mixing CM/iface client-server connection establishing for current code
2) it's easier to extend protocol getting assertion failures on impossible flows before

Reply:
ok, pls add code comment

Reply:
added https://github.com/openucx/ucx/pull/5244/commits/5a2118b4d9871d2c12db6f3eded9ffd4e08a7221#diff-5e97c1d955a563dd705187d5ded3b527R652